src/ELF/Symbol.cpp

```diff
@@ -214,8 +214,9 @@ bool Symbol::is_imported() const {
   // An import must not be defined in a section
   bool is_imported = shndx() == static_cast<uint16_t>(SYMBOL_SECTION_INDEX::SHN_UNDEF);
 
-  // An import must not have an address
-  is_imported = is_imported && value() == 0;
+  // An import must not have an address but mips and powerpc put an addres for
```

Comment:
Would you have an example/sample for this?

Reply:
Yes of course, https://transfer.ci-yow.com/d?id=RcdhAdQ3vsAuWeX, this is one the function imported "strstr" which have an st_value.

I have some better example if you want but I can't share you here.

Reply:
No thanks it's a good sample.
I see the issue, my only concern is that your change drop the condition for **all** the architectures, not only MIPS/PPC. 
On the other hand, the targeted architecture is not accessible from the `Symbol` class. 
I need to think about how we can fix this design issue.

Reply:
I am agree, I would like to limited this drop this check only for these two architectures but as you said, I didn't find how to do it without a big refactorisation of you code...

But when I check how some technique to identify imported function, and how the loader do, it should look symbols and resolve *undefined symbols*. So the main meaning check should be:
```
  bool is_imported = shndx() == static_cast<uint16_t>(SYMBOL_SECTION_INDEX::SHN_UNDEF);
```
the others is for safety but they should not add bug.

I let you see what you prefer and if you would like to do a big factorisation to get access of ARCH by symbols, I could help you to dev :).